internal/cloudapi/v2/server.go

```diff
@@ -445,22 +446,79 @@ func (s *Server) enqueueKojiCompose(taskID uint64, server, name, version, releas
 }
 
 func serializeManifest(ctx context.Context, manifestSource *manifest.Manifest, workers *worker.Server, depsolveJobID, containerResolveJobID, ostreeResolveJobID, manifestJobID uuid.UUID, seed int64) {
-	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	// prepared to become a config variable
+	const depsolveTimeout = 5
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*depsolveTimeout)
 	defer cancel()
 
-	// wait until job is in a pending state
-	var token uuid.UUID
+	jobResult := &worker.ManifestJobByIDResult{
+		Manifest: nil,
+		ManifestInfo: worker.ManifestInfo{
+			OSBuildComposerVersion: common.BuildVersion(),
+		},
+	}
+
 	var dynArgs []json.RawMessage
 	var err error
+	token := uuid.Nil
 	logWithId := logrus.WithField("jobId", manifestJobID)
+
+	defer func() {
+		// token == uuid.Nil indicates that no worker even started processing
+		if token == uuid.Nil {
+			if jobResult.JobError != nil {
+				// set all jobs to "failed"
+				// osbuild job will fail as dependency
+				jobs := map[string]uuid.UUID{
```

Comment:
I wonder if it's ok to use a map here, since that means the loop below could be out of order. I'm not sure if there's a problem with failing the manifest job before the resolve jobs.